File: ruby/send_list.rb
#!/usr/bin/env ruby
# إرسال رسالة بقائمة تفاعلية عبر واتساب
# Send interactive list message via WhatsApp
# واٹس ایپ کے ذریعے انٹرایکٹو لسٹ پیغام بھیجیں

require "net/http"
require "uri"
require "json"
require "base64"

# إعدادات الاتصال - Connection settings - رابطہ کی ترتیبات
APP_KEY = "your_app_key"
API_SECRET = "your_api_secret"
PROJECT_ID = "your_project_id"
RECIPIENT = "9665XXXXXXXX"

def send_list_message
  # إرسال رسالة قائمة - Send list message - لسٹ پیغام بھیجیں
  url = URI("https://api-users.4jawaly.com/api/v1/whatsapp/#{PROJECT_ID}")
  auth_string = "#{APP_KEY}:#{API_SECRET}"
  auth_b64 = Base64.strict_encode64(auth_string)

  payload = {
    path: "global",
    params: {
      url: "messages",
      method: "post",
      data: {
        messaging_product: "whatsapp",
        to: RECIPIENT,
        type: "interactive",
        interactive: {
          type: "list",
          header: { type: "text", text: "قائمة الخدمات" },
          body: { text: "اختر الخدمة المطلوبة من القائمة أدناه" },
          footer: { text: "4Jawaly Services" },
          action: {
            button: "عرض القائمة",
            sections: [
              {
                title: "الخدمات الأساسية",
                rows: [
                  { id: "svc_sms", title: "خدمة الرسائل النصية", description: "إرسال رسائل SMS للعملاء" },
                  { id: "svc_whatsapp", title: "خدمة واتساب", description: "إرسال رسائل واتساب تفاعلية" }
                ]
              },
              {
                title: "الدعم الفني",
                rows: [
                  { id: "support_ticket", title: "فتح تذكرة دعم", description: "تواصل مع فريق الدعم الفني" }
                ]
              }
            ]
          }
        }
      }
    }
  }

  http = Net::HTTP.new(url.host, url.port)
  http.use_ssl = true
  http.verify_mode = OpenSSL::SSL::VERIFY_PEER

  request = Net::HTTP::Post.new(url)
  request["Content-Type"] = "application/json"
  request["Authorization"] = "Basic #{auth_b64}"
  request.body = payload.to_json

  begin
    response = http.request(request)
    puts "رمز الاستجابة / Response code / جوابی کوڈ: #{response.code}"
    puts "\nالاستجابة / Response / جواب:\n#{JSON.pretty_generate(JSON.parse(response.body))}"
  rescue Net::OpenTimeout, Net::ReadTimeout, SocketError => e
    puts "خطأ في الاتصال / Connection error / رابطہ میں خرابی: #{e.message}"
  rescue JSON::ParserError => e
    puts "خطأ في تحليل الاستجابة / Response parse error / جواب کی تشریح میں خرابی: #{e.message}"
    puts "النص الخام / Raw text / خام متن: #{response&.body}"
  end
end

send_list_message if __FILE__ == $PROGRAM_NAME

File: kotlin/src/main/kotlin/SendList.kt
// إرسال رسالة بقائمة تفاعلية عبر واتساب
// Send interactive list message via WhatsApp
// واٹس ایپ کے ذریعے انٹرایکٹو لسٹ پیغام بھیجیں

import okhttp3.MediaType.Companion.toMediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody
import org.json.JSONArray
import org.json.JSONObject
import java.util.Base64

// إعدادات الاتصال - Connection settings - رابطہ کی ترتیبات
private const val APP_KEY = "your_app_key"
private const val API_SECRET = "your_api_secret"
private const val PROJECT_ID = "your_project_id"
private const val RECIPIENT = "9665XXXXXXXX"

fun main() {
    // إرسال رسالة قائمة - Send list message - لسٹ پیغام بھیجیں
    try {
        val url = "https://api-users.4jawaly.com/api/v1/whatsapp/$PROJECT_ID"
        val authString = "$APP_KEY:$API_SECRET"
        val authB64 = Base64.getEncoder().encodeToString(authString.toByteArray(Charsets.UTF_8))

        // بناء أقسام القائمة - Build list sections - لسٹ سیکشن بنائیں
        val section1Rows = JSONArray()
            .put(JSONObject().apply {
                put("id", "svc_sms")
                put("title", "خدمة الرسائل النصية")
                put("description", "إرسال رسائل SMS للعملاء")
            })
            .put(JSONObject().apply {
                put("id", "svc_whatsapp")
                put("title", "خدمة واتساب")
                put("description", "إرسال رسائل واتساب تفاعلية")
            })

        val section2Rows = JSONArray()
            .put(JSONObject().apply {
                put("id", "support_ticket")
                put("title", "فتح تذكرة دعم")
                put("description", "تواصل مع فريق الدعم الفني")
            })

        val sections = JSONArray()
            .put(JSONObject().apply {
                put("title", "الخدمات الأساسية")
                put("rows", section1Rows)
            })
            .put(JSONObject().apply {
                put("title", "الدعم الفني")
                put("rows", section2Rows)
            })

        val interactive = JSONObject().apply {
            put("type", "list")
            put("header", JSONObject().apply {
                put("type", "text")
                put("text", "قائمة الخدمات")
            })
            put("body", JSONObject().put("text", "اختر الخدمة المطلوبة من القائمة أدناه"))
            put("footer", JSONObject().put("text", "4Jawaly Services"))
            put("action", JSONObject().apply {
                put("button", "عرض القائمة")
                put("sections", sections)
            })
        }

        val data = JSONObject().apply {
            put("messaging_product", "whatsapp")
            put("to", RECIPIENT)
            put("type", "interactive")
            put("interactive", interactive)
        }
        val params = JSONObject().apply {
            put("url", "messages")
            put("method", "post")
            put("data", data)
        }
        val payload = JSONObject().apply {
            put("path", "global")
            put("params", params)
        }

        val client = OkHttpClient()
        val request = Request.Builder()
            .url(url)
            .addHeader("Content-Type", "application/json")
            .addHeader("Authorization", "Basic $authB64")
            .post(payload.toString().toRequestBody("application/json".toMediaType()))
            .build()

        client.newCall(request).execute().use { response ->
            println("رمز الاستجابة / Response code / جوابی کوڈ: ${response.code}")
            println("\nالاستجابة / Response / جواب:\n${response.body?.string() ?: ""}")
        }
    } catch (e: Exception) {
        // خطأ في الاتصال - Connection error - رابطہ میں خرابی
        println("خطأ / Error / خرابی: ${e.message}")
        e.printStackTrace()
    }
}
